web/packages/api/src/forKusama.ts

```diff
@@ -0,0 +1,800 @@
+    totalFeeInDot: bigint
+}
+
+export enum Direction {
+    ToKusama,
+    ToPolkadot,
+}
+
+function resolveInputs(
+    registry: AssetRegistry,
+    tokenAddress: string,
+    sourceParaId: number,
+    destParaId: number
+) {
+    const sourceParachain = registry.parachains[sourceParaId.toString()]
+    if (!sourceParachain) {
+        throw Error(`Could not find ${sourceParaId} in the asset registry.`)
+    }
+    const destParachain = registry.kusama?.parachains[destParaId.toString()]
+    if (!destParachain) {
+        throw Error(`Could not find ${destParaId} in the asset registry.`)
+    }
+
+    const sourceAssetMetadata = registry.parachains[sourceParaId].assets[tokenAddress.toLowerCase()]
+    if (!sourceAssetMetadata) {
+        throw Error(`Token ${tokenAddress} not registered on source asset hub.`)
+    }
+    const destAssetMetadata =
+        registry.kusama?.parachains[destParaId].assets[tokenAddress.toLowerCase()]
+    if (!destAssetMetadata) {
+        throw Error(`Token ${tokenAddress} not registered on destination asset hub.`)
+    }
+
+    return { sourceAssetMetadata, destAssetMetadata, sourceParachain }
+}
+
+export async function getDeliveryFee(
+    sourceAssetHub: ApiPromise,
+    direction: Direction,
+    registry: AssetRegistry,
+    defaultFee?: bigint
+): Promise<DeliveryFee> {
+    const feeStorageKey = xxhashAsHex(":XcmBridgeHubRouterBaseFee:", 128, true)
+    const feeStorageItem = await sourceAssetHub.rpc.state.getStorage(feeStorageKey)
+    let leFee = new BN((feeStorageItem as Codec).toHex().replace("0x", ""), "hex", "le")
+
+    let defaultFeeConfiguredInRuntime: bigint
+    if (direction == Direction.ToPolkadot) {
+        defaultFeeConfiguredInRuntime = 10_602_492_378n // .0106KSM
+    } else {
+        defaultFeeConfiguredInRuntime = 333_794_429n // 0.033 DOT
+    }
+    let xcmBridgeFee: bigint
+    if (leFee.eqn(0)) {
+        console.warn("Asset Hub onchain XcmBridgeHubRouterBaseFee not set. Using default fee.")
+        xcmBridgeFee = defaultFee ?? defaultFeeConfiguredInRuntime
+    } else {
+        xcmBridgeFee = BigInt(leFee.toString())
+    }
+
+    let forwardedXcm = buildTransferToKusamaExportXCM(
+        sourceAssetHub.registry,
+        DOT_LOCATION,
+        dotLocationOnKusamaAssetHubLocation(),
+        erc20Location(registry.ethChainId, "0x0000000000000000000000000000000000000000"), // actual token location doesn't matter here, just weighing the message
+        xcmBridgeFee,
+        xcmBridgeFee,
+        registry.assetHubParaId,
+        340282366920938463463374607431768211455n,
+        "0x0000000000000000000000000000000000000000000000000000000000000000",
+        "0x0000000000000000000000000000000000000000000000000000000000000000"
+    )
+
+    let bridgeHubDeliveryFee = await calculateDeliveryFee(
+        sourceAssetHub,
+        registry.bridgeHubParaId,
+        forwardedXcm
+    )
+
+    // In either DOT or KSM
+    let totalFee = xcmBridgeFee + bridgeHubDeliveryFee
```

Comment:
For the P-K bridge, it seems to be a bit more complicated than our P-E bridge. It looks like they take the message-fee [here](https://github.com/paritytech/polkadot-sdk/blob/e44b89fb7ca3385f314803c733ad97b26cd14e9f/bridges/modules/xcm-bridge-hub-router/src/lib.rs#L324-L325) into consideration. So it's `base_fee + message_fee + delivery_fee` in total.